cluster-autoscaler/debuggingsnapshot/debugging_snapshot_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debuggingsnapshot
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+func TestBasicSetterWorkflow(t *testing.T) {
+	snapshot := &DebuggingSnapshotImpl{}
+	pod := []*framework.PodInfo{
+		{
+			Pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "Pod1",
+				},
+				Spec: v1.PodSpec{
+					NodeName: "testNode",
+				},
+			},
+		},
+	}
+	node := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "testNode",
+		},
+	}
+
+	nodeInfo := &framework.NodeInfo{
+		Pods:             pod,
+		Requested:        &framework.Resource{},
+		NonZeroRequested: &framework.Resource{},
+		Allocatable:      &framework.Resource{},
+		Generation:       0,
+	}
+
+	var nodeGroups []*framework.NodeInfo
+	nodeGroups = append(nodeGroups, nodeInfo)
+	nodeGroups[0].SetNode(node)
+	timestamp := time.Now().In(time.UTC)
+	snapshot.SetNodeGroupInfo(nodeGroups)
+	snapshot.SetEndTimestamp(timestamp)
+	op, err := snapshot.GetOutputBytes()
+	assert.False(t, err)
+
+	var parsed map[string]interface{}
+	er := json.Unmarshal(op, &parsed)
+	assert.NoError(t, er)
+	pNodeInfo := parsed["NodeList"].([]interface{})[0].(map[string]interface{})
```

Comment:
Can you check if these type assertions are ok so that if they fail, we can get a test error instead of panic?